ext/standard/tests/array/max_int_float_optimisation.phpt

```diff
@@ -0,0 +1,101 @@
+--TEST--
+Check max() optimisation for int and float types
+--SKIPIF--
+<?php if (PHP_INT_SIZE != 8) die("skip this test is for 64bit platform only"); ?>
+--FILE--
+<?php
+
+echo "Start as int optimisation:\n";
+var_dump(max(
+    10, 5, 3, 2
+));
+var_dump(max(
+    2, 3, 5, 10
+));
+var_dump(max(
+    10, 5, 3.5, 2
+));
+var_dump(max(
+    2, 3.5, 5, 10
+));
+var_dump(max(
+    10, 5, "3", 2
+));
+var_dump(max(
+    2, "3", 5, 10
+));
+var_dump(max(
+    2, 3, "15", 10
+));
+echo "Check that int not representable as float works:\n";
+var_dump(max(
+    PHP_INT_MIN+1, PHP_INT_MIN, PHP_INT_MIN*2)
+);
+var_dump(max(
+    PHP_INT_MAX-1, PHP_INT_MAX, PHP_INT_MAX*2)
+);
+// Has INF
+var_dump(max(
+    PHP_INT_MAX-1, PHP_INT_MAX, PHP_INT_MAX**20)
+);
+
+echo "Start as float optimisation:\n";
+var_dump(max(
+    10.5, 5.5, 3.5, 2.5
+));
+var_dump(max(
+    2.5, 3.5, 5.5, 10.5
+));
+var_dump(max(
+    10.5, 5.5, 3, 2.5
+));
+var_dump(max(
+    2.5, 3, 5.5, 10.5
+));
+var_dump(max(
+    10.5, 5.5, "3.5", 2.5
+));
+var_dump(max(
+    2.5, "3.5", 5.5, 10.5
+));
+var_dump(max(
+    2.5, 3.5, "15.5", 10.5
+));
+echo "Check that int not representable as float works:\n";
+var_dump(max(
+    PHP_INT_MIN*2, PHP_INT_MIN, PHP_INT_MIN+1)
+);
```

Comment:
NIT: Couldn't all of those be on a single line. Btw this one is a bit inconsistent formatting from the above ones as well...